src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,190 @@
 """
 from __future__ import print_function
 
-from command_utils import ExecutableCommand
+import os
+import getpass
+
+from command_utils import DaosCommand, CommandWithParameters
 from command_utils import BasicParameter, FormattedParameter
+from general_utils import get_file_path
+from avocado.utils import process
 
 
-class DmgCommand(ExecutableCommand):
+class DmgCommand(DaosCommand):
     """Defines a object representing a dmg (or daos_shell) command."""
 
     def __init__(self, path):
         """Create a dmg Command object."""
         super(DmgCommand, self).__init__("/run/dmg/*", "daos_shell", path)
+        self.format = self.DmgFormatSubCommand()
+        self.prepare = self.DmgPrepareSubCommand()
 
-        self.request = BasicParameter("{}")
-        self.action = BasicParameter("{}")
-
-        # daos_shell options
         self.hostlist = FormattedParameter("-l {}")
         self.hostfile = FormattedParameter("-f {}")
         self.configpath = FormattedParameter("-o {}")
         self.insecure = FormattedParameter("-i", None)
+        self.debug = FormattedParameter("-d", None)
+        self.json = FormattedParameter("-j", None)
+        self.subcommand_list = {
+            "format": self.format,
+            "prepare": self.prepare,
+        }
+
+    def __str__(self):
```

Comment:
The combining of the DaosCommand command string + command arguments + request + action command + action command arguments feels like it will be generic enough to be included in the DaosCommand object

Reply:
I'll remove.